src/lib/ngsi/ContextElement.cpp

```diff
@@ -154,7 +157,7 @@ void ContextElement::fill(const struct ContextElement& ce)
   attributeDomainName.fill(ce.attributeDomainName);
   contextAttributeVector.fill((ContextAttributeVector*) &ce.contextAttributeVector);
   domainMetadataVector.fill((MetadataVector*) &ce.domainMetadataVector);
-  providingApplication = ce.providingApplication;  
+  providingApplicationList = ce.providingApplicationList;  // FIXME P10: not sure if this is the right way of doing, taking into account how the same is done for contextAttributeVector in its own fill() method...
```

Comment:
Is the `=` operator clever enough to actually copying vector contents? Looking to other similar structures (attributeList) I think you have encapsulated it in a separated class with speciall method for copying (with pass through all the vector, pushing back element from one to other). Maybe I should do the same?

I'm to used to high level languages in which this is not a problem at all that I'm a bit unsure at C++ :)